mock/http/server.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2021 Telefonica Cybersecurity & Cloud Tech SL
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package http
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Server type for the HTTP mock server.
+type Server struct {
+	Port         int
+	mockRequests MockRequests
+	logger       *logrus.Entry
+}
+
+// NewServer creates an instance of Server.
+func NewServer(port int) *Server {
+	return &Server{
+		Port:         port,
+		mockRequests: MockRequests{},
+		logger:       logrus.WithField("mock", "http"),
+	}
+}
+
+// Start the HTTP mock server.
+// Note that it block the current goroutine with http.ListenAndServe function.
+func (s *Server) Start() error {
+	http.HandleFunc("/_mock/requests", s.handleMockRequest)
+	http.HandleFunc("/", s.handle)
+	addr := fmt.Sprintf(":%d", s.Port)
+	s.logger.Infof("Starting server at '%s'", addr)
+	return http.ListenAndServe(addr, nil)
+}
+
+func (s *Server) handleMockRequest(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case http.MethodPost:
+		var mockRequest MockRequest
```

Comment:
I found problems with the use of var when passed to a `Decode` library, exactly with `yaml.Decode(config)`
It complains about not been able to access memory to store read data in config.
To isolate you from these problems I would suggest to change it to have structure already allocated:
```
	mockRequest := &MockRequest{}
     	if err := json.NewDecoder(r.Body).Decode(mockRequest); err != nil {
```

Reply:
It's exactly the same.
Note that it's important to pass a pointer to Decode method